### PR TITLE
Add support for retention when deleting ApigeeOrganization

### PR DESCRIPTION
**Pull request by xuchenma**
This PR solves https://github.com/hashicorp/terraform-provider-google/issues/12219

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
apigee: added support for specifying retention when deleting `google_apigee_organization`
```

### CODE REVIEW

**Comment by xuchenma** [comment]
Hi Sam, this API isn't public yet (will be there later this week) so the tests won't pass for now.
Meanwhile I'm not sure if this is the correct way to add a query parameter that only works for the DELETE call. Could you check please? Thanks!

File: mmv1/products/apigee/api.yaml [hunk 1]
```diff
@@ -64,6 +64,20 @@ objects:
...
+          Optional. This setting is applicable only for organizations that are soft-deleted (i.e., BillingType
+          is not EVALUATION). It controls how long Organization data will be retained after the initial delete
+          operation completes. During this period, the Organization may be restored to its last known state.
+          After this period, the Organization will no longer be able to be restored.
+        values:
+          - "DELETION_RETENTION_UNSPECIFIED"
+          - "MINIMUM"
+        default_value: :DELETION_RETENTION_UNSPECIFIED
+        required: false
+        input: true
```
Comment (slevenick) [inline mmv1/products/apigee/api.yaml:79]: We don't want input here, as this value is only sent during the delete request. Input forces Terraform to recreate the resource if this value changes

File: mmv1/products/apigee/api.yaml [hunk 2]
```diff
@@ -64,6 +64,20 @@ objects:
...
+        url_param_only: true
```
Comment (slevenick) [inline mmv1/products/apigee/api.yaml:80]: I believe this will be sent with create requests as well if implemented like this. Likely not what we want

**Review by slevenick — CHANGES_REQUESTED** [review]
It may take some experimentation to find what works for this API. Do you have access to a staging environment?

**Comment by xuchenma** [comment]
> It may take some experimentation to find what works for this API. Do you have access to a staging environment?

No but with TF_LOG=debug I'm able to verify if the HTTP request is made correctly, so I'm still able to verify if this PR works.

I tried removing "input: true" and "url_param_only: true", but then "retention" is treated as a data field (rather than a query param) in the HTTP request, which returns a 400 bad request:
```
    "message": "Invalid JSON payload received. Unknown name \"retention\" at 'organization': Cannot find field.",
```

 I'm wondering what is the proper way to treat "retention" as a query param only for DELETE call?

**Comment by slevenick** [comment]
This is a very unusual field, but yeah query param only seems right.

It definitely cannot be input: true though

**Comment by xuchenma** [comment]
It seems that with "url_param_only: true" the ApigeeOrganization was created properly, but the delete call was made without the "retention" param:

---[ REQUEST ]---------------------------------------
DELETE /v1/organizations/tf-teste0zf16pcki?alt=json HTTP/1.1
Host: apigee.googleapis.com
User-Agent: Terraform/0.13.6 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google-beta/acc
Content-Type: application/json
Accept-Encoding: gzip

**Comment by xuchenma** [comment]
The solution here is to add "delete_url" to the resource, set "url_param_only" to true to the field, and skip status check (since LRO will expire when Org is deleted).

File: mmv1/products/apigee/api.yaml [hunk 3]
```diff
@@ -36,8 +36,10 @@ objects:
     name: 'Organization'
     base_url: 'organizations'
     create_url: 'organizations?parent=projects/{{project_id}}'
+    delete_url: 'organizations/{{name}}?retention={{retention}}'
     self_link: 'organizations/{{name}}'
     async: !ruby/object:Api::OpAsync
+      actions: ['create']
```
Comment (slevenick) [inline mmv1/products/apigee/api.yaml:42]: We want update in here too, only want to omit delete

File: mmv1/products/apigee/terraform.yaml [hunk 4]
```diff
@@ -47,6 +47,16 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
       # Resource creation race
       skip_vcr: true
+    - !ruby/object:Provider::Terraform::Examples
+      name: "apigee_organization_retention_test"
+      primary_resource_id: "org"
+      test_env_vars:
+        org_id: :ORG_ID
+        billing_account: :BILLING_ACCT
+      skip_docs: true
+      min_version: beta
```
Comment (slevenick) [inline mmv1/products/apigee/terraform.yaml:57]: Is the feature beta-only? That may cause some issues if the parameter needs to exist in the URL string, as that is shared across beta and GA

Comment (xuchenma) [reply mmv1/products/apigee/terraform.yaml:57]: No, right now Apigee has no beta API, but running the full Terraform acceptance tests for this feature depends on a beta-only resource.

**Review by slevenick — COMMENTED** [review]
This looks good at this point. Let me know when the API rolls out and I can test this

**Comment by xuchenma** [comment]
> This looks good at this point. Let me know when the API rolls out and I can test this

This API should be live now. Thanks!

**Review by slevenick — APPROVED** [review]
Tests passed when triggered manually

**Comment by xuchenma** [comment]
Thank you Sam!

**Pull request merged**